docs/contracts/storage-layout-specifier.rst

```diff
@@ -0,0 +1,61 @@
+.. index:: ! storage layout specifier, ! layout at
+
+.. _storage-layout-specifier:
+
+************************
+Storage Layout Specifier
+************************
+
+Contracts can define an arbitrary base slot for its own storage.
+The contract's state variables will be stored from the specified slot
+instead of the default slot zero.
+This can be done by using ``layout at <base-slot-expression>`` in the header of
+a contract definition, either after or before the inheritance specifier.
+It can be given at most once.
+The ``base-slot-expression`` must be an expression consisting of number literals
+that can be evaluated at compile time and yield in a rational value in the range of ``uint256``.
```

Comment:
Perhaps another section describing that the max number of available storage slots will be (uint256::max - user defined base slot address + 1), and that a revert will be triggered in case an attempt is made to exceed the maximum number of available slots.

Reply:
Not a revert. A compilation error.